vignettes/visits_periods.Rmd

```diff
@@ -0,0 +1,273 @@
+---
+title: "Visit and Period Variables"
+output:
+  rmarkdown::html_vignette:
+vignette: >
+  %\VignetteIndexEntry{Visit and Period Variables}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+The derivation of visit variables like `AVISIT`, `AVISITN`, `AWLO`, `AWHI`, ...
+or period, subperiod, or phase variables like `APERIOD`, `TRT01A`, `TRT02A`,
+`ASPER`, `PHSDTM`, `PHEDTM`, ... is highly study-specific. Therefore admiral can
+not provide functions which derive these variables. However, for common
+scenarios like visit assignments based on time windows or deriving BDS period
+variables from ADSL period variables functions are provided which support those
+derivations.
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(tibble)
+library(dplyr)
+library(lubridate)
+```
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE}
+library(admiraldev)
+```
+
+# Visit variables (`AVISIT`, `AVISITN`, `AWLO`, `AWHI`, ...) {#visits}
+
+The most common ways of deriving `AVISIT` and `AVISITN` are:
+
+- The variables are set to the collected visits (`VISIT` and `VISITNUM`).
+- The variables are set based on time windows.
+
+The former one can be done by calling `mutate()` like in the vignettes and the
+template scripts.
+
+For the latter one a (study specific) reference dataset needs to be created
+which provides for each visit the start and end day (`AWLO` and `AWHI`) and the
+values of other visit related variables (`AVISIT`, `AVISITN`, `AWTARGET`, ...).
```

Comment:
do you think worth to mention anything about the periods being non-overlapping here or is that obvious?

Reply:
I think that's obvious.

Reply:
ok